share/rocm/cmake/header_template.h.in

```diff
@@ -5,21 +5,17 @@
 #ifndef @include_guard@
 #define @include_guard@
 
-#if defined(ROCM_NO_WRAPPER_HEADER_WARNING) || defined(ROCM_@ITEM_GUARD@_GAVE_WARNING)
+#if defined(ROCM_NO_WRAPPER_HEADER_ERROR) || defined(ROCM_@ITEM_GUARD@_GAVE_ERROR)
 /* include file */
 #include "@file_rel_path@"
 #else
-/* give warning */
-#if defined(_MSC_VER)
-#pragma message(": warning:This file is deprecated. Use the header file from @header_location@ by using #include <@correct_include@>")
-#elif defined(__GNUC__)
-#warning "This file is deprecated. Use the header file from @header_location@ by using #include <@correct_include@>"
-#endif
+/* give error */
+#error "This file is deprecated. Use the header file from @header_location@ by using #include <@correct_include@>"
```

Comment:
Do we need special handling for microsoft here?

Reply:
No. [`#error`](https://en.cppreference.com/w/c/preprocessor/error) is a standard C89 preprocessor directive.